src/Features/Core/Portable/Completion/Providers/ImportCompletionProvider/ExtensionMethodImportCompletionHelper.ExtensionMethodSymbolComputer.cs

```diff
@@ -0,0 +1,539 @@
+                        // to the receiverTypeSymbol, then no need to proceed methods from this group..
+                        continue;
+                    }
+
+                    // This is also affected by the symbol resolving issue mentioned above, which means in case referenced projects
+                    // are targeting different framework, we will miss extension methods with any framework type in their signature from those projects.
+                    var isFirstMethod = true;
+                    foreach (var methodInCurrentCompilation in methodSymbols.Select(s => SymbolFinder.FindSimilarSymbols(s, SemanticModel.Compilation).FirstOrDefault()).WhereNotNull())
+                    {
+                        if (isFirstMethod)
+                        {
+                            isFirstMethod = false;
+
+                            // We haven't seen this receiver type yet. Try to check by reducing one extension method
+                            // to the given receiver type and save the result.
+                            if (!cachedResult)
+                            {
+                                // If this is the first symbol we retrived from current compilation,
+                                // try to check if we can apply it to given receiver type, and save result to our cache.
+                                // Since method symbols are grouped by their declared receiver type, they are either all matches to the receiver type
+                                // or all mismatches. So we only need to call ReduceExtensionMethod on one of them.
+                                var reducedMethodSymbol = methodInCurrentCompilation.ReduceExtensionMethod(ReceiverTypeSymbol);
+                                cachedResult = reducedMethodSymbol != null;
+                                CheckedReceiverTypes[declaredReceiverTypeInCurrentCompilation] = cachedResult;
+
+                                // Now, cachedResult being false means method doesn't match the receiver type,
+                                // stop processing methods from this group.
+                                if (!cachedResult)
+                                {
+                                    break;
+                                }
+                            }
+                        }
+
+                        if (SemanticModel.IsAccessible(Position, methodInCurrentCompilation))
+                        {
+                            builder.Add(methodInCurrentCompilation);
+                        }
+                    }
+                }
+
+                return builder.ToImmutable();
+            }
+
+            private ImmutableArray<IMethodSymbol> GetExtensionMethodsForSymbolsFromSameCompilation(
+                MultiDictionary<ITypeSymbol, IMethodSymbol> matchingMethodSymbols,
+                CancellationToken cancellationToken)
```

Comment:
on the fence if we should have this method and the one aboe.  they are effetively the same, except you do the FindSimilarSymbols call in teh above one.  we could just have one method and have it decide to do that or not depending on which compilation we're currently looking at.

Reply:
I started with a single method, but eventual came to this because 
(1) it provides more clarity on these two different cases.
(2) those methods when separated, are fairly simple